influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_filter.rs

```diff
@@ -306,6 +307,51 @@ pub async fn read_filter_periods_multi_field_predicate2() {
     .await
 }
 
+#[tokio::test]
+pub async fn read_filter_multi_data_frame() {
+    // there will be 1000 points per frame, so
+    // this number of points will return 3 frames
+    // with the last containing just one point.
+    let num_points: i64 = 2001;
+    let input_lines: Vec<String> = (0..num_points)
+        .map(|i| {
+            format!(
+                "h2o,tagk0=tagv0 f0={} {}",
+                i,
+                1_000_000_000 + i * 1_000_000_000
+            )
+        })
+        .collect();
+    let input_lines: Vec<&str> = input_lines.iter().map(String::as_ref).collect();
+
+    let mut expected = vec!["SeriesFrame, tags: _field=f0,_measurement=h2o,tagk0=tagv0, type: 0"];
+    let ts_vec = (0..num_points)
+        .map(|i| 1_000_000_000 + i * 1_000_000_000)
+        .collect::<Vec<_>>();
+    let values_vec = (0..num_points).collect::<Vec<_>>();
+    let mut data_frames: Vec<String> = vec![];
+    for (ts_chunk, v_chunk) in ts_vec.chunks(1000).zip(values_vec.chunks(1000)) {
+        let ts_str = ts_chunk.iter().map(|ts| ts.to_string()).join(", ");
+        let v_str = v_chunk.iter().map(|v| v.to_string()).join(",");
+        data_frames.push(format!(
+            "FloatPointsFrame, timestamps: [{}], values: \"{}\"",
+            ts_str, v_str
+        ));
+    }
+    data_frames
+        .iter()
+        .for_each(|line| expected.push(line.as_ref()));
+
```

Comment:
Given the point of this test is that the frames are broken up into 3 messages, perhaps an additional assert here would make the intent more clear (so someone doesn't accidentally update this test in some future refactoring)

```suggestion

    // Response should be broken into 3 frames
    assert_eq!(data_frames.len(), 3);

```